fg/apps/factory/views/factory.py

```diff
@@ -81,3 +86,263 @@ def view_factoryorder_parts(request, uuid, subset=None):
+                    dist = Distribution.objects.create(uuid=uuid.UUID(distEntry['uuid']),
+                                                       description=distEntry['description'],
+                                                       name=distEntry['name'])
+
+            # Create each well
+            for wellEntry in entry['wells']:
+                sampleEntry = wellEntry['sample']
+                partEntry = sampleEntry['part']
+                authorEntry = partEntry['author']               
+                organismEntry = wellEntry['organism']
+
+                # Author is associated with a part                
+                try:
+                    author = Author.objects.get(uuid=authorEntry['uuid'])
+                except Author.DoesNotExist:
+                    author = Author.objects.create(uuid=uuid.UUID(authorEntry['uuid']),
+                                                   email=authorEntry['email'],
+                                                   affiliation=authorEntry['affiliation'],
+                                                   orcid=authorEntry['orcid'])
+
+                # Create the Part (retrieve based on gene id, not uuid)
+                try:
+                    part = Part.objects.get(gene_id=partEntry['gene_id'])
+                except Part.DoesNotExist:
+                    counts['parts']+=1
+                    part = Part.objects.create(uuid=uuid.UUID(partEntry['uuid']),
+                                               description=partEntry['description'],
+                                               status=partEntry['status'],
+                                               original_sequence=partEntry['original_sequence'],
+                                               genbank=partEntry['genbank'],
+                                               synthesized_sequence=partEntry['synthesized_sequence'],
+                                               full_sequence=partEntry['full_sequence'],
+                                               vector=partEntry['vector'],
+                                               primer_forward=partEntry['primer_forward'],
+                                               primer_reverse=partEntry['primer_reverse'],
+                                               barcode=partEntry['barcode'],
+                                               translation=partEntry['translation'],
+                                               author=author)
+
+                # Tags are associated with a part
+                for tagEntry in partEntry['tags']:
+                    name = tagEntry.get('tag')
+                    if name is not None:
+                        tag, created = Tag.objects.get_or_create(tag=tagEntry)
+                        part.tags.add(tag)
+                        if created:
+                            counts['tags']+=1
+
+                part.save()
+
+                # Generate the sample (we don't also import derived from)
```

Comment:
If not overly difficult, I think this might be important for tracking of the history of objects.

Reply:
Thinking this over, it's more complicated than just recusing samples. It's also including parts that are associated for each one. This gets very hairy very quickly... you'll be importing a lot of (what I think you call physicals) that aren't actually physical.

Reply:
Any sample which is derived from another sample should share the same part (by definition). However, I can see where that isn't a constraint that is enforced in the system.

Samples are kind of a mix of physicals and virtuals. They do exist (somewhere), but their primary use is denote provenance (when was this thing actually sequenced and where?)

For example, I've been distributed a plate from someone, and I want to know when a certain part was sequenced. How many culture steps has it been since then?

Reply:
To be clear - you want to export all recursive samples AND parts, even if the "physical" entities are not actually physical.

Reply:
I suppose if the constraint isn't in place, yes recurse on all samples AND parts, though in reality I don't think there will ever be a time where you'll actually need to recursively export parts, since by definition of what a derived sample is, you'll always be hitting the same part.